gwcs/tests/test_geometry.py

```diff
@@ -92,3 +112,57 @@ def test_cartesian_to_spherical_mixed_Q(cart_to_spher, x, y, z):
         cart_to_spher(x, y, z)
     assert (arg_err.value.args[0] == "All arguments must be of the same type "
             "(i.e., quantity or not).")
+
+
+@pytest.mark.parametrize('wrap_at', ['1', 180., True, 180j, [180], -180, 0])
+def test_c2s2c_wrong_wrap_type(spher_to_cart, cart_to_spher, wrap_at):
+    err_msg = "'wrap_lon_at' must be an integer number: 180 or 360"
+    with pytest.raises(ValueError) as arg_err:
+        geometry.SphericalToCartesian(wrap_lon_at=wrap_at)
+    assert arg_err.value.args[0] == err_msg
+
+    with pytest.raises(ValueError) as arg_err:
+        spher_to_cart.wrap_lon_at = wrap_at
+    assert arg_err.value.args[0] == err_msg
+
+    with pytest.raises(ValueError) as arg_err:
+        geometry.CartesianToSpherical(wrap_lon_at=wrap_at)
+    assert arg_err.value.args[0] == err_msg
+
+    with pytest.raises(ValueError) as arg_err:
+        cart_to_spher.wrap_lon_at = wrap_at
+    assert arg_err.value.args[0] == err_msg
+
+
+def test_cartesian_spherical_asdf(spher_to_cart, cart_to_spher):
```

Comment:
There is a nice test helper function in asdf called `assert_roundtrip_tree`.  It works like this:

```python
from asdf.tests.helpers import assert_roundtrip_tree

assert_roundtrip_tree({'c2s': cart_to_spher, 's2c': spher_to_cart})
```

The function will create an AsdfFile, assign your tree, write to a temporary file, read it back in, and assert that the deserialized tree matches the original tree.

Reply:
@eslavich After some thinking, I think I would still prefer current test followed by a mini coordinate round-trip test. However, it seems to me comparing the tree add additional testing. Do you think I should *add* the above `assert_roundtrip_tree` to current test *instead* of replacing?

Reply:
(we chatted about this on Slack and came to the conclusion that it's a good idea to use assert_roundtrip_tree, since it makes an especially thorough set of comparisons)